gtsam/slam/ReadMe.md

```diff
@@ -0,0 +1,64 @@
+# SLAM Factors
+
+## GenericProjectionFactor (defined in ProjectionFactor.h)
+
+Non-linear factor for a constraint derived from a 2D measurement. 
+The calibration is assumed known and passed in the constructor.
+The main building block for visual SLAM.
+
+Templated on
+- `POSE`, default `Pose3`
+- `LANDMARK`, default `Point3`
+- `CALIBRATION`, default `Cal3_S2`
+
+## SmartFactors
+
+These are "structure-less" factors, i.e., rather than introducing a new variable for an observed 3D point or landmark, a single factor is created that provides a multi-view constraint on several poses and/or cameras.
+
+### SmartFactorBase
+
+This is the base class for smart factors, templated on a `CAMERA` type.
+It has no internal point, but it saves the measurements, keeps a noise model, and an optional sensor pose.
+
+### SmartProjectionFactor
+
+Also templated on `CAMERA`. Triangulates a 3D point and keeps an estimate of it around.
+This factor operates with monocular cameras, and is used to optimize the camera pose
+*and* calibration for each camera, and requires variables of type `CAMERA` in values.
+
+If the calibration is fixed use `SmartProjectionPoseFactor` instead!
+
+
+### SmartProjectionPoseFactor
+
+Derives from `SmartProjectionFactor` but is templated on a `CALIBRATION` type, setting `CAMERA = PinholePose<CALIBRATION>`.
+This factor assumes that the camera calibration is fixed and the same for all cameras involved in this factor.
+The factor only constrains poses.
+
+If the calibration should be optimized, as well, use `SmartProjectionFactor` instead!
+
+### SmartProjectionFactorP
+
+Same as `SmartProjectionPoseFactor`, except:
+- it is templated on `CAMERA`, i.e., it allows cameras beyond pinhole;
+- it admits a different calibration for each measurement, i.e., it can model a multi-camera system;
+- it allows multiple observations from the same pose/key, again, to model a multi-camera system.
+
+TODO: DimPose and ZDim are hardcoded. Copy/paste from `SmartProjectionPoseFactor`. Unclear what the use case is.
+
+### RegularImplicitSchurFactor
```

Comment:
I would put this part in a separate subsection. The factors below are less likely to be relevant to the user (at least for VSLAM applications).

Reply:
Done - but something you should know is that the RegularImplicitSchurFactor is very fast with CG...

Reply:
for full SfM or also for fixed lag smoothing? (for the latter, the linear system is small, so maybe Cholesky is still faster?)

Reply:
Yes, maybe you're right